edge/pkg/metamanager/process.go

```diff
@@ -477,3 +491,41 @@ func (m *metaManager) runMetaManager() {
 		}
 	}()
 }
+
+func podMatchByNodeNameOrHostIP(message *model.Message, nodeName, hostIP string) (isMatched bool) {
+	content := message.GetContent()
+	var k8sResourceMap map[string]interface{}
+	var spec map[string]interface{}
+	var status map[string]interface{}
+	if _, ok := content.(map[string]interface{}); ok {
+		k8sResourceMap = content.(map[string]interface{})
+	}
+	if k8sResourceMap != nil {
+		if _, ok := k8sResourceMap["spec"].(map[string]interface{}); ok {
+			spec = k8sResourceMap["spec"].(map[string]interface{})
+		}
+		if _, ok := k8sResourceMap["status"].(map[string]interface{}); ok {
+			status = k8sResourceMap["status"].(map[string]interface{})
+		}
+	}
+
+	if spec != nil {
+		if _, ok := spec["nodeName"]; ok {
+			nodeNameFromResource := spec["nodeName"].(string)[0:]
+			isMatched = nodeNameFromResource == nodeName
+			if isMatched {
+				return
+			}
+		}
+	}
+	if status != nil {
+		if _, ok := status["hostIP"]; ok {
+			hostIPFromResource := status["hostIP"].(string)[0:]
+			isMatched = hostIPFromResource == hostIP
+			if isMatched {
+				return
+			}
+		}
+	}
+	return
```

Comment:
We have the resourceType in the message, can identify pod using it. Besides, just nodename is enough? :)